### PR TITLE
Enabled CLI timing.

### PR DESCRIPTION
**Pull request by avitebskiy**

### CODE REVIEW

File: common/src/OpenSpaceNet.cpp [hunk 1]
```diff
@@ -261,6 +261,10 @@ void OpenSpaceNet::process()
 void OpenSpaceNet::setProgressDisplay(boost::shared_ptr<deepcore::ProgressDisplay> display)
 {
     pd_ = display;
```
Comment (rddesmond) [inline common/src/OpenSpaceNet.cpp]: Move `display` into `pd_`.

File: common/src/OpenSpaceNet.cpp [hunk 2]
```diff
@@ -625,16 +629,9 @@ FileFeatureSink::Ptr OpenSpaceNet::initFeatureSink()
 
 void OpenSpaceNet::startProgressDisplay()
 {
-    if(!pd_ || args_.quiet) {
-        return;
+    if(pd_ && !args_.quiet) {
```
Comment (rddesmond) [inline common/src/OpenSpaceNet.cpp]: since this is all that remains, consider inlining the remaining 3 lines

File: cli/src/CliProcessor.cpp [hunk 3]
```diff
@@ -272,8 +272,11 @@ void CliProcessor::setupArgParsing(int argc, const char* const* argv)
 void CliProcessor::startOSNProcessing()
 {
     OpenSpaceNet osn(std::move(osnArgs));
-    pd_ = boost::make_shared<ConsoleProgressDisplay>();
-    osn.setProgressDisplay(pd_);
+
+    auto pd = boost::make_shared<ConsoleProgressDisplay>();
+    osn.setProgressDisplay(pd);
+    pd->enableTiming("Detecting");
```
Comment (rddesmond) [inline cli/src/CliProcessor.cpp]: move this to `setProgressDisplay` when we set the categories.

File: cli/src/CliProcessor.cpp [hunk 4]
```diff
@@ -272,8 +272,11 @@ void CliProcessor::setupArgParsing(int argc, const char* const* argv)
 void CliProcessor::startOSNProcessing()
 {
     OpenSpaceNet osn(std::move(osnArgs));
-    pd_ = boost::make_shared<ConsoleProgressDisplay>();
-    osn.setProgressDisplay(pd_);
+
+    auto pd = boost::make_shared<ConsoleProgressDisplay>();
+    osn.setProgressDisplay(pd);
```
Comment (rddesmond) [inline cli/src/CliProcessor.cpp]: I believe you can `move` `pd` here (given that you move the def of enable timing downstream).

Comment (avitebskiy) [reply cli/src/CliProcessor.cpp]: I cannot. I need `osn.setProgressDisplay` to initialize categories before I can enable timing, so I have to hold on to the pointer.

File: cli/src/CliProcessor.cpp (hunk 3 above)
Comment (avitebskiy) [reply cli/src/CliProcessor.cpp]: I cannot. `ProgressDisplay` doesn't have timing. It's only in `ConsoleProgressDisplay`

File: common/src/OpenSpaceNet.cpp (hunk 2 above)
Comment (avitebskiy) [reply common/src/OpenSpaceNet.cpp]: No, I'll keep it here. This way there's room for logic. There's enough inlined crap in `OpenSpaceNet::process` already.

**Pull request merged**